main.nf

```diff
@@ -267,12 +277,57 @@ if ( !params.skip_basecalling ) {
     fi
     """
   }
+} else if ( params.skip_basecalling && params.skip_demultiplexing ){
+
+  if (params.input) { 
+    ch_input_files = Channel.fromPath(params.input, checkIfExists: true)
+  } else { 
+    exit 1, "Please specify a valid run directory to perform rename process!" 
+  }
+
+  process rename_barcode {
+    publishDir path: "${params.outdir}/renamed_barcodes", mode:'copy'
+
+    input:
+    file fastq_files from ch_input_files
+    file csv_file from ch_input_csv.ifEmpty([])
+      
+    output:
+    file "fastq/*.fastq.gz" into ch_fastq, ch_for_seqkit
+    file "rename.log" into ch_log_rename
+
+    script:
+    """
+    mkdir fastq
+    fastqdir=\$PWD
+    cd $fastq_files
+    if [ "\$(find . -type d -name "barcode*" )" != "" ]
+    then
+      for dir in barcode*/
+      do
+        dir=\${dir%*/}
+        cat \$dir/*.fastq.gz > \$fastqdir/fastq/\$dir.fastq.gz
```

Comment:
what if the fastq files are *.fastq and not *.fastq.gz - this is normally the case when we run MinKNOW. We need to run pigz, I already included it in the docker image.